chain-abci/src/app/app_init/validator_state.rs

```diff
@@ -35,12 +33,265 @@ pub struct ValidatorState {
     /// FIXME: fairness -- use lastcommitinfo for stats
     proposer_stats: BTreeMap<StakedStateAddress, u64>,
 
+    after_unbond_delete: DeleteScheduleMap,
+
     /// various lookups for handling validator updates, rewards, punishments
-    #[codec(skip)]
     pub validator_state_helper: ValidatorStateHelper,
 }
 
+#[derive(PartialEq, Debug, Clone, Default)]
+struct DeleteScheduleMap {
+    ordered_by_time: BTreeMap<(Timespec, StakedStateAddress), TendermintValidatorAddress>,
+    lookup_time: BTreeMap<(TendermintValidatorAddress, StakedStateAddress), Timespec>,
```

Comment:
I think `StakedStateAddress ` should be unique, can be used as key alone?

Reply:
For the re-join cases within unbond period, this may not be the case:
1. node operator unbonds -> (tm_a1, a1) leaves
2. (deposit or reward); node operator rejoins with (tm_a2, a1)
3. node operator unbonds -> (tm_a2, a1) leaves
4. (deposit or reward); node operator rejoins with (tm_a1, a1)